src/data/finset/noncomm_prod.lean

```diff
@@ -318,4 +365,38 @@ begin
     exact comm_gf x (mem_insert_self x s) y (mem_insert_of_mem hy) this, }
 end
 
+section finite_pi
+
+variables {ι : Type*} [fintype ι] [decidable_eq ι] {M : ι → Type*} [∀ i, monoid (M i)]
+variables (x : Π i, M i)
+
+@[to_additive]
+lemma noncomm_prod_mul_single :
+  univ.noncomm_prod (λ i, pi.mul_single i (x i))
+  (λ i _ j _, by { by_cases h : i = j, { rw h }, { apply pi.mul_single_commute i j h } })
```

Comment:
TODO: When are the laptop, check if this is just `pi.mul_single_apply_commute` now